app/models/spree/credit_card_decorator.rb

```diff
@@ -25,4 +28,16 @@ def has_payment_profile? # rubocop:disable Style/PredicateName
   def save_requested_by_customer?
     !!@save_requested_by_customer
   end
+
+  private
+
+  def default_missing?
+    user.credit_cards.where(is_default: true).none?
+  end
+
+  def ensure_default
+    return unless user
```

Comment:
isn't `user` mandatory for a credit card? I feel like this guard close shouldn't be needed.

Reply:
No, `user` it not mandatory on `credit_card`. A user can checkout using a credit card without opting to store it against their account. In this case a new card object is created (as the source for the payment) but it is not directly linked to the user's account.